Source/WebCore/platform/network/CacheValidation.cpp

```diff
@@ -68,6 +68,9 @@ static constexpr ASCIILiteral headerPrefixesToIgnoreAfterRevalidation[] = {
 
 static inline bool shouldUpdateHeaderAfterRevalidation(const String& header)
 {
+    if (header.startsWithIgnoringASCIICase("content-security-"_s))
+        return true;
+
     for (auto& headerToIgnore : headersToIgnoreAfterRevalidation) {
```

Comment:
Idea: We should consider using a better data structure for these headers to avoid doing this loop (especially if there are more than a handful).

Reply:
The arrays are just a few lines up. There's 13 headers we do the equality check for and 3 prefixes we check against in the next loop. At this small size a hash set's overhead might negate any theoretical improvement.